pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/instance/InstanceAssignmentDriver.java

```diff
@@ -104,17 +106,15 @@ public InstancePartitions assignInstances(String tierName, List<InstanceConfig>
   private InstancePartitions getInstancePartitions(String instancePartitionsName,
       InstanceAssignmentConfig instanceAssignmentConfig, List<InstanceConfig> instanceConfigs,
       @Nullable InstancePartitions existingInstancePartitions,
-      @Nullable InstancePartitions preConfiguredInstancePartitions, @Nullable Boolean minimizeDataMovementOverride) {
+      @Nullable InstancePartitions preConfiguredInstancePartitions, Enablement minimizeDataMovementEnablement) {
     String tableNameWithType = _tableConfig.getTableName();
 
     // minimizeDataMovement might be set back to false within InstanceTagPoolSelector and InstancePartitionSelector
     // if existingInstancePartitions is null.
     boolean minimizeDataMovement =
-        minimizeDataMovementOverride == null ? instanceAssignmentConfig.isMinimizeDataMovement()
-            : minimizeDataMovementOverride;
-    LOGGER.info("Starting {} instance assignment for table {}, original minimizeDataMovement: {}, "
-            + "overriding with minimizeDataMovementOverride: {}", instancePartitionsName, tableNameWithType,
-        instanceAssignmentConfig.isMinimizeDataMovement(), minimizeDataMovementOverride);
+        minimizeDataMovementEnablement.isEnabled(instanceAssignmentConfig::isMinimizeDataMovement);
+    LOGGER.info("Starting {} instance assignment for table: {} with minimizeDataMovement: {}", instancePartitionsName,
```

Comment:
nit: shall we also log the `instanceAssignmentConfig.isMinimizeDataMovement()` value for debugging purposes?

Reply:
Sure